### PR TITLE
Add GET publish endpoint [RHELDST-14412]

### PR DESCRIPTION
**Pull request by negillett**
This commit implements a new endpoint to retrieve a publish from the service database. No items are returned with the publish as there is currently no use case for them and implementation would be non-trival.

### CODE REVIEW

File: exodus_gw/routers/publish.py [hunk 1]
```diff
@@ -287,3 +287,61 @@ def commit_publish(
...
+                        "links": {
+                            "self": "/prod/publish/497f6eca-6276-4993-bfeb-53cbbbba6f08",
+                            "commit": "/prod/publish/497f6eca-6276-4993-bfeb-53cbbbba6f08/commit",
+                        },
+                        "items": [],
+                    }
+                }
+            },
+        },
+        422: {
```
Comment (rohanpm) [inline exodus_gw/routers/publish.py:313]: Seems like this should be 404, not 422.

File: tests/routers/test_publish.py [hunk 2]
```diff
@@ -779,3 +779,73 @@ def test_commit_no_publish(auth_header):
...
+    """
+
+    with TestClient(app) as client:
+        r = client.get(
+            "/test/publish/%s" % fake_publish.id,
+            headers=auth_header(roles=["test-publisher"]),
+        )
+
+    # It should have failed
+    assert r.status_code == 422
```
Comment (rohanpm) [inline tests/routers/test_publish.py:847]: Should be 404 and not 422.

**Review by rohanpm — CHANGES_REQUESTED** [review]

File: exodus_gw/routers/publish.py (hunk 1 above)
Comment (negillett) [reply exodus_gw/routers/publish.py:313]: @rohanpm, the Jira specified 422 but 404 makes sense too.

Comment (rohanpm) [reply exodus_gw/routers/publish.py]: I see, I think the JIRA was wrong, will comment there.

**Review by rohanpm — APPROVED** [review]

File: exodus_gw/routers/publish.py [hunk 3]
```diff
@@ -287,3 +287,62 @@ def commit_publish(
...
+    This is achieved using the noload() query option.
+    """
+
+    db_publish = (
+        db.query(models.Publish)
+        .options(noload("items"))
+        .filter(
+            # Since sub-environments share some resources, filter for env as well.
+            models.Publish.id == publish_id,
+            models.Publish.env == env.name,
```
Comment (rohanpm) [inline exodus_gw/routers/publish.py:338]: The other endpoints aren't using 'env' during lookup.  Noticing that made me realize that the other endpoints are buggy, and they should also all be using 'env' like this one is.  Filed a new issue RHELDST-15336.

Comment (negillett) [reply exodus_gw/routers/publish.py:338]: 👍🏻 Thanks

**Pull request merged**